### ISSUE TITLE
[Bug]: ERROR in × Module not found: Can't resolve './src/main.tsx' in

### ISSUE THREAD

**Issue by hasnum-stack** [web-infra-dev/rspack#8446]
### System Info

  System:
    OS: Linux 3.10 CentOS Linux 7 (Core)
    CPU: (16) x64 Intel(R) Xeon(R) Silver 4210 CPU @ 2.20GHz
    Memory: 7.36 GB / 23.37 GB
    Container: Yes
    Shell: 4.2.46 - /bin/bash
  Binaries:
    Node: 16.16.0 - /opt/node-v16.16.0/bin/node
    npm: 8.11.0 - /opt/node-v16.16.0/bin/npm
    pnpm: 9.13.0 - /home/jenkins/.local/share/pnpm/pnpm
  npmPackages:
    @rspack/cli: ^1.1.1 => 1.1.1
    @rspack/core: ^1.1.1 => 1.1.1
    @rspack/plugin-react-refresh: 1.0.0 => 1.0.0

### Details

ERROR in × Module not found: Can't resolve './src/main.tsx' in '/data1/jenkins_dir/testrsbuild/123123'

### Reproduce Steps

pnpm create rspack@latest
pnpm i
pnpm build

**Comment by hasnum-stack** [comment]
感觉是centos 7的问题 ...

**Comment by theninthsky** [comment]
Same issue on Cloudflare Pages with rspack v1.1.1:

```
10:38:51.399 | > rspack build --mode production
-- | --
10:38:51.400 |  
10:38:52.276 | ERROR in × Module not found: Can't resolve './src' in '/opt/buildhome/repo'
10:38:52.277 | help: Found module './src/index.tsx'. However, it's not possible to request this module without the extension
10:38:52.277 | if its extension was not listed in the `resolve.extensions`. Here're some possible solutions:
10:38:52.278 |  
10:38:52.278 | 1. add the extension `".tsx"` to `resolve.extensions` in your rspack configuration
10:38:52.278 | 2. use './src/index.tsx' instead of './src'
10:38:52.278 |  
10:38:52.279 | Rspack compiled with 1 error in 158 ms
10:38:52.307 | Failed: Error while executing user command. Exited with error code: 1
10:38:52.316 | Failed: build command exited with code: 1
10:38:53.047 | Failed: error occurred while running build command
```

On my windows machine it builds just fine.

**Comment by chenjiahan** [comment]
Fixed in Rspack v1.1.2
